src/python/pants/backend/python/target_types_test.py

```diff
@@ -237,19 +245,34 @@ def assert_inferred(address: Address, *, expected: Address | None) -> None:
 
     # Warn if there's ambiguity, meaning we cannot infer.
     caplog.clear()
-    assert_inferred(Address("project", target_name="ambiguous"), expected=None)
-    assert len(caplog.records) == 1
+    if python_infer_unowned_dependency_behavior == "error":
+        with pytest.raises(ExecutionError) as ambiguous_error:
+            assert_inferred(Address("project", target_name="ambiguous"), expected=None)
+        assert isinstance(ambiguous_error.value.wrapped_exceptions[0], UnownedDependencyError)
```

Comment:
this should be tabbed, I think

Reply:
(same below)

Reply:
making assertions on the exception caught by pytest is done after the `with` block. It gets me every time because it feels like a scope escape, especially since the `with` block is often used for resource lifecycle management.

Reply:
Riiiiiiiiiiiiight because line 250 _raises an exception_. Gahh OK. One point you 😛